tests/test_setup_external_storage.py

```diff
@@ -1,52 +1,38 @@
 import boto3
 import pytest
 import synapseclient
-from pyarrow import fs, parquet
+from pyarrow import fs
 
+from src.scripts.setup_external_storage import setup_external_storage
 
-@pytest.fixture
-def test_parquet_folder():
-    """TODO: Replace with production parquet folder synapse ID
-    or save as external environment variable"""
-    parquet_folder = "syn51079888"
-    return parquet_folder
 
+@pytest.fixture()
+def test_synapse_folder_id(pytestconfig):
+    return pytestconfig.getoption("test_synapse_folder_id")
 
-@pytest.fixture
-def test_synapse_client():
-    """Returns a synapse client from credentials stored in SSM"""
-    aws_session = boto3.session.Session(profile_name="default", region_name="us-east-1")
-    ssm_parameter = "synapse-recover-auth"
-    if ssm_parameter is not None:
-        ssm_client = aws_session.client("ssm")
-        token = ssm_client.get_parameter(Name=ssm_parameter, WithDecryption=True)
-        test_synapse_client = synapseclient.Synapse()
-        test_synapse_client.login(authToken=token["Parameter"]["Value"])
-    else:  # try cached credentials
-        test_synapse_client = synapseclient.login()
-    return test_synapse_client
 
+@pytest.fixture()
+def test_ssm_parameter(pytestconfig):
+    return pytestconfig.getoption("test_ssm_parameter")
 
-def test_setup_external_storage_success(test_parquet_folder, test_synapse_client):
+
+def test_setup_external_storage_success(test_synapse_folder_id, test_ssm_parameter):
     """This test tests that it can get the STS token credentials and view and list the
-    parquet files in the S3 bucket location to verify that it has access"""
+    files in the S3 bucket location to verify that it has access"""
+    test_synapse_client = setup_external_storage.get_synapse_client(
+        ssm_parameter=test_ssm_parameter, aws_session=boto3
+    )
     # Get STS credentials
     token = test_synapse_client.get_sts_storage_token(
-        entity=test_parquet_folder, permission="read_only", output_format="json"
+        entity=test_synapse_folder_id, permission="read_only", output_format="json"
     )
```

Comment:
What happens if we try to get an STS token for a storage location which is not properly configured for use with Synapse? My guess is that Synapse will throw an exception and everything below this line is doing more to test the `pyarrow` package than it is to test our `setup_external_storage` script.

Reply:
Yes, we could leave out the lines after this, because this is the line that essentially tests the STS access